src/main/java/io/cdap/plugin/switchcase/route/RoutingSwitch.java

```diff
@@ -0,0 +1,348 @@
+    @Nullable
+    private final String defaultPort;
+
+    @Name(NULL_HANDLING_PROPERTY_NAME)
+    @Description("Determines the way to handle records whose value for the routing field is null. Such records can " +
+      "either be skipped (\"Skip\"), sent to a specific port (\"Send to null port\"), or sent to the error port " +
+      "(\"Send to error port\"). By default, such records are sent to the error port.")
+    @Nullable
+    private final String nullHandling;
+
+    @Name("nullPort")
+    @Description("Determines the port to which records with null values for the field to split on are routed to. " +
+      "This is only used if default handling is set to \"Send to null port\". Defaults to 'Null'.")
+    @Nullable
+    private final String nullPort;
+
+    Config(@Nullable String routingField, @Nullable String portSpecification, @Nullable String defaultHandling,
+           @Nullable String defaultPort, @Nullable String nullHandling, @Nullable String nullPort) {
+      this.routingField = routingField;
+      this.portSpecification = portSpecification;
+      this.defaultHandling = defaultHandling == null ? DefaultHandling.ERROR_PORT.value : defaultHandling;
+      this.defaultPort = defaultPort == null ? DEFAULT_PORT_NAME : defaultPort;
+      this.nullHandling = nullHandling == null ? NullHandling.ERROR_PORT.value : nullHandling;
+      this.nullPort = nullPort == null ? DEFAULT_NULL_PORT_NAME : nullPort;
+    }
+
+    private void validate(Schema inputSchema, FailureCollector collector) throws IllegalArgumentException {
+      // TODO: Add validation code for mode after adding jexl mode
+      Optional<DefaultHandling> handling = DefaultHandling.fromValue(defaultHandling);
+      if (!handling.isPresent()) {
+        collector.addFailure(String.format("Unsupported default handling value %s", DEFAULT_HANDLING_PROPERTY_NAME),
+                             String.format("Please specify one of %s", Joiner.on(",").join(DefaultHandling.values())))
+          .withConfigProperty(DEFAULT_HANDLING_PROPERTY_NAME);
+      }
+
+      // TODO: Add validation for jexl mode after adding jexl support
+
+      if (routingField == null || routingField.isEmpty()) {
+        collector.addFailure("Field to split is required when routing mode is basic.",
+                             "Please provide the field to split on").withConfigProperty(ROUTING_FIELD_PROPERTY_NAME);
+      }
+      Schema.Field field = inputSchema.getField(routingField);
+      if (field == null) {
+        collector.addFailure(String.format("Routing field %s not found in the input schema", routingField),
+                             "Please provide a field that exists in the input schema")
+          .withConfigProperty(ROUTING_FIELD_PROPERTY_NAME);
+        throw collector.getOrThrowException();
+      }
+      Schema fieldSchema = field.getSchema();
+      Schema.Type type = fieldSchema.isNullable() ? fieldSchema.getNonNullable().getType() : fieldSchema.getType();
```

Comment:
we should use displayname

Reply:
I thought displayname was only for messages (logs and exceptions and validation)? This type is actually used in a ``if`` check. Do we need it as well?

Reply:
Ah yea thats right. But in the error message, please use displayname. that will make sure logicaltypes are also handled correctly for error message.